### PR TITLE
feature added: support for api RolePermissionsBoundary

### PR DESCRIPTION
**Pull request by macnev2013**
This PR adds support for `PutRolePermissionsBoundary` and `DeleteRolePermissionsBoundary`.
Fixed the response template `GET_ROLE_TEMPLATE` for RolePermissionBoundary support.
Added test case for `PutRolePermissionsBoundary` and `DeleteRolePermissionsBoundary`.

This PR should fix the issue in localstack: [localstack/localstack#2888](https://github.com/localstack/localstack/issues/2888)

### CODE REVIEW

**Comment by macnev2013** [comment]
Hey @bblommers . Can you help me understand why tests are failing?
I haven't made any changes in the test cases which are failing.

It would be great if you can have a quick look at it.
Thanks.

File: moto/iam/models.py [hunk 1]
```diff
@@ -1292,6 +1292,23 @@ def update_role(self, role_name, role_description, max_session_duration):
         role.max_session_duration = max_session_duration
         return role
 
+    def put_role_permissions_boundary(self, role_name, permissions_boundary):
+        if permissions_boundary and not self.policy_arn_regex.match(
+            permissions_boundary
+        ):
+            raise RESTError(
```
Comment (bblommers) [inline moto/iam/models.py:1299]: Hi @macnev2013, can you add a unit test to verify that this error is thrown?

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @macnev2013, just have a single comment.
Don't worry about that test - it's an intermittent failure, and a second run usually succeeds.

File: moto/iam/models.py (hunk 1 above)
Comment (macnev2013) [reply moto/iam/models.py:1299]: Hey @bblommers . Thanks for the update.

Looks like we already have a test cases for it.
[https://github.com/spulec/moto/blob/master/tests/test_iam/test_iam.py#L2380-L2386](https://github.com/spulec/moto/blob/master/tests/test_iam/test_iam.py#L2380-L2386)

Comment (bblommers) [reply moto/iam/models.py:1299]: That's for the `create_role`-method though. It's the same logic, but we should still test the negative scenario against `put_role_permissions_boundary`

Comment (nevilmacwan1) [reply moto/iam/models.py:1299]: Hey @bblommers . I have added test case for `put_role_permissions_boundary`.
Please let me know if anything else is required.

Comment (macnev2013) [reply moto/iam/models.py:1299]: Hey @bblommers. I have added test case for put_role_permissions_boundary.
Please let me know if anything else is required.

**Review by bblommers — APPROVED** [review]
LGTM. Thanks @macnev2013!

**Pull request merged**